doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,487 @@
+
+Tutorial #1 - Reachability Monitor
+==================================
+
+In our first tutorial we are going to build a program that will calculate the reachability information in the network.
+User will update link status between nodes through CLI and our program will calculate the rest.
+The purpose of this program is to provide a simple interface to a network operator and return information regarding which node can reach which.
+Thus, the only concern of network operator is to update link status without having to worry about doing those calculations themself.
+We are going to test our program with the network consisting of four nodes: Menlo Park, Santa Barbara, Los Angeles, and Salt Lake City.
+However, nothing prevents us from using the network comprised of hundreds of nodes.
+
+Let's start with defining our program in the DDlog language.
+In our program we need to supply source node, destination node, and a link status between them.
+Now we are ready to define our input relation :code:`Links`.
+Let's create a file called :code:`t1_reachability_monitor.dl` and add there a following line representing input relation for our reachability monitor:
+
+.. code-block::
+    
+    input relation Links(src: string, dst: string, link_status: bool)
+
+However, for the reachability monitor to be useful it also should provide some output and not just ingest the data.
+We want our program to output a list of nodes that are currently reachable based on the information in the relation :code:`Links`.
+Thus, let's add an output relation :code:`ConnectedNodes` and provide rules how to calculate records in this relation *(mind the presence of the dot at the end of each rule)*.
+To do that we are going to add few more lines to the :code:`t1_reachability_monitor.dl` so it will look as follows:
+
+.. code-block::
+
+    input relation Links(src: string, dst: string, link_status: bool)
+    output relation ConnectedNodes(src: string, dst: string)
+
+    ConnectedNodes(src, dst) :- Links(src, dst, true).
+    ConnectedNodes(src, dst) :- ConnectedNodes(src, intermediate_node),
+                                Links(intermediate_node, dst, true).
+
+Let's briefly discuss the rule to calculate :code:`ConnectedNodes` relation.
+First, we say that any two directly connected nodes such that the :code:`link_status` between them is :code:`true` belong to the :code:`ConnectedNodes` relation.
+Furthermore, any two nodes are considered connected if there is path from the source node to the destination node such that all directly connected nodes along the path also have the :code:`link_status` between them to be :code:`true`.
+Take a minute here to realize how easy it is to write such rule in DDlog.
+While these rules may not require much effort in other programming languages, the distinct feature of DDlog is that the computing is done incrementally!
+Isn't is amazing?

```

Comment:
Your users haven't seen yet how the incremental aspect works, so I don't know if they will appreciate it at this point.